contracts/modules/Experimental/TransferManager/LockupVolumeRestrictionTM.sol

```diff
@@ -229,164 +238,208 @@ contract LockupVolumeRestrictionTM is ITransferManager {
-     * @param totalAmount Total amount of locked up tokens
+     * @param _userAddress Address whom lockup is being applied
+     * @param _lockupAmount Amount that needs to be locked
+     * @param _lockUpPeriodSeconds Total period of lockup (seconds)
+     * @param _releaseFrequencySeconds How often to release a tranche of tokens (seconds)
      */
-    function _checkLockUpParams(uint lockUpPeriodSeconds, uint releaseFrequencySeconds, uint totalAmount) internal view {
-        require(lockUpPeriodSeconds != 0, "lockUpPeriodSeconds cannot be zero");
-        require(releaseFrequencySeconds != 0, "releaseFrequencySeconds cannot be zero");
-        require(totalAmount != 0, "totalAmount cannot be zero");
+    function _checkLockUpParams(
+        address _userAddress,
+        uint256 _lockupAmount,
+        uint256 _lockUpPeriodSeconds,
+        uint256 _releaseFrequencySeconds
+    ) 
+        internal 
+        view 
+    {   
+        require(_userAddress != address(0), "Invalid address");
+        require(_lockUpPeriodSeconds != 0, "lockUpPeriodSeconds cannot be zero");
+        require(_releaseFrequencySeconds != 0, "releaseFrequencySeconds cannot be zero");
+        require(_lockupAmount != 0, "lockupAmount cannot be zero");
 
         // check that the total amount to be released isn't too granular
         require(
-            totalAmount % ISecurityToken(securityToken).granularity() == 0,
+            _lockupAmount % ISecurityToken(securityToken).granularity() == 0,
             "The total amount to be released is more granular than allowed by the token"
         );
 
         // check that releaseFrequencySeconds evenly divides lockUpPeriodSeconds
         require(
-            lockUpPeriodSeconds % releaseFrequencySeconds == 0,
+            _lockUpPeriodSeconds % _releaseFrequencySeconds == 0,
             "lockUpPeriodSeconds must be evenly divisible by releaseFrequencySeconds"
         );
 
-        // check that totalPeriods evenly divides totalAmount
-        uint totalPeriods = lockUpPeriodSeconds.div(releaseFrequencySeconds);
+        // make sure the amount to be released per period is not too granular for the token
+        uint256 totalPeriods = _lockUpPeriodSeconds.div(_releaseFrequencySeconds);
+        uint256 amountPerPeriod = _lockupAmount.div(totalPeriods);
         require(
-            totalAmount % totalPeriods == 0,
-            "The total amount being locked up must be evenly divisible by the number of total periods"
+            amountPerPeriod.mul(totalPeriods) == _lockupAmount,
+            "lockup amount should be completely divisible by the amount per period"
         );
-
-        // make sure the amount to be released per period is not too granular for the token
-        uint amountPerPeriod = totalAmount.div(totalPeriods);
         require(
             amountPerPeriod % ISecurityToken(securityToken).granularity() == 0,
```

Comment:
Wondering if it really matters if this is not true (amount per period is at a different granularity than the token)? Main concern is that it could be a bit confusing for issues  to have to ensure all these conditions are met (which could be hard if granularity is 1, and you're locking a prime number amount of tokens...)

Reply:
Suppose if granularity is 1 for the token and I passed prime no. as the lockup Amount with a 4 period then if I remove this check it will create the lockup for the 1 token and doesn't allow to user to withdrawal until full lockup gets passed. so prime nos are blocked for token whose granularity is one but allowed whose granularity is 18.

Reply:
gotcha what you are trying to say here.